# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: 71670cedaf26
# Full Hash: 71670cedaf26214f3bcb8562a445cf1ef24f4db6
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-06-30 09:36:48
# Regressor Bug: 1714914
# File Overlap Count: 1
# Description:
#   Bug 1714914 - Make editor classes treat offset in parent node as uint32_t r=m_kato
#   
#   It should be treated as `uint32_t` since DOM API does so.  However, there are
#   some exceptions:
#   
# ==============================================================================

diff -r f5dcc2d793d6 -r 71670cedaf26 editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Tue Jun 29 22:00:09 2021 +0000
+++ b/editor/libeditor/HTMLEditor.h	Wed Jun 30 07:07:28 2021 +0000
@@ -867,10 +867,9 @@
    * Helper routines for font size changing.
    */
   enum class FontSize { incr, decr };
-  MOZ_CAN_RUN_SCRIPT nsresult RelativeFontChangeOnTextNode(FontSize aDir,
-                                                           Text& aTextNode,
-                                                           int32_t aStartOffset,
-                                                           int32_t aEndOffset);
+  MOZ_CAN_RUN_SCRIPT nsresult
+  RelativeFontChangeOnTextNode(FontSize aDir, Text& aTextNode,
+                               uint32_t aStartOffset, uint32_t aEndOffset);
 
   MOZ_CAN_RUN_SCRIPT nsresult SetInlinePropertyOnNode(nsIContent& aNode,
                                                       nsAtom& aProperty,
@@ -1625,7 +1624,7 @@
    */
   [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult
   HandleInsertParagraphInHeadingElement(Element& aHeader, nsINode& aNode,
-                                        int32_t aOffset);
+                                        uint32_t aOffset);
 
   /**
    * HandleInsertParagraphInListItemElement() handles insertParagraph command
@@ -1639,7 +1638,7 @@
    */
   [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult
   HandleInsertParagraphInListItemElement(Element& aListItem, nsINode& aNode,
-                                         int32_t aOffset);
+                                         uint32_t aOffset);
 
   /**
    * InsertParagraphSeparatorAsSubAction() handles insertPargraph commad